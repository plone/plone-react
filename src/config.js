/**
 * Config.
 * @module config
 */
import { defaults } from 'lodash';

export default defaults(
  {},
  {
    host: process.env.HOST,
    port: process.env.PORT,
    apiPath: process.env.API_PATH,
  },
  {
    host: 'localhost',
    port: '4300',
    apiPath: 'http://localhost:8081/db/web',
  },
);
