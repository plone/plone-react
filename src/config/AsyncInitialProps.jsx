// import { getBaseUrl } from '@plone/volto/helpers';
// import config from '@plone/volto/registry';
// import {
//   getBreadcrumbs,
//   getContent,
//   getNavigation,
//   getTypes,
//   getWorkflow,
// } from '@plone/volto/actions';

const AsyncInitialProps = [
  {
    route: {
      path: '/**',
    },
    asyncProps: [
      // {
      //   key: 'breadcrumbs',
      //   promise: ({ location, store: { dispatch } }) => {
      //     return (
      //       __SERVER__ &&
      //       dispatch(getBreadcrumbs(getBaseUrl(location.pathname)))
      //     );
      //   },
      // },
      // {
      //   key: 'content',
      //   promise: ({ location, store: { dispatch } }) =>
      //     __SERVER__ && dispatch(getContent(getBaseUrl(location.pathname))),
      // },
      // {
      //   key: 'navigation',
      //   promise: ({ location, store: { dispatch } }) =>
      //     __SERVER__ &&
      //     dispatch(
      //       getNavigation(
      //         getBaseUrl(location.pathname),
      //         config.settings.navDepth,
      //       ),
      //     ),
      // },
      // {
      //   key: 'types',
      //   promise: ({ location, store: { dispatch } }) =>
      //     __SERVER__ && dispatch(getTypes(getBaseUrl(location.pathname))),
      // },
      // {
      //   key: 'workflow',
      //   promise: ({ location, store: { dispatch } }) =>
      //     __SERVER__ && dispatch(getWorkflow(getBaseUrl(location.pathname))),
      // },
    ],
  },
];

export default AsyncInitialProps;
