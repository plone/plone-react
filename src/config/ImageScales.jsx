export const imageScales = {
  large: 768,
  preview: 400,
  mini: 200,
  thumb: 128,
  tile: 64,
  icon: 32,
  listing: 16,
};
