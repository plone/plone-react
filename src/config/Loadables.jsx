import loadable from '@loadable/component';

export const loadables = {
  prettierStandalone: loadable.lib(() => import('prettier/standalone')),
  prettierParserHtml: loadable.lib(() => import('prettier/parser-html')),
  prismCore: loadable.lib(() => import('prismjs/components/prism-core')),
  toastify: loadable.lib(() => import('react-toastify'), {
    resolveComponent: (lib) => lib.toast,
  }),
  reactSelect: loadable.lib(() => import('react-select')),
};
