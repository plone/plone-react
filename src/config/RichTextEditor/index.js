import ToHTMLRenderers, { options as ToHTMLOptions } from './ToHTML';
import { extendedBlockRenderMap, blockStyleFn, listBlockTypes } from './Blocks';
import plugins, { inlineToolbarButtons } from './Plugins';
import FromHTMLCustomBlockFn from './FromHTML';

export {
  extendedBlockRenderMap,
  blockStyleFn,
  listBlockTypes,
  FromHTMLCustomBlockFn,
  inlineToolbarButtons as richTextEditorInlineToolbarButtons,
  plugins as richTextEditorPlugins,
  ToHTMLRenderers,
  ToHTMLOptions,
};
