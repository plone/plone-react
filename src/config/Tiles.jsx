import { defineMessages } from 'react-intl';

import ViewTitleTile from '@plone/volto/components/manage/Tiles/Title/View';
import ViewDescriptionTile from '@plone/volto/components/manage/Tiles/Description/View';
import ViewTextTile from '@plone/volto/components/manage/Tiles/Text/View';
import ViewImageTile from '@plone/volto/components/manage/Tiles/Image/View';
import ViewVideoTile from '@plone/volto/components/manage/Tiles/Video/View';
import ViewHeroImageLeftTile from '@plone/volto/components/manage/Tiles/HeroImageLeft/View';
import ViewMapTile from '@plone/volto/components/manage/Tiles/Maps/View';
import ViewHTMLTile from '@plone/volto/components/manage/Tiles/HTML/View';
import ViewTableTile from '@plone/volto/components/manage/Tiles/Table/View';

import EditTitleTile from '@plone/volto/components/manage/Tiles/Title/Edit';
import EditDescriptionTile from '@plone/volto/components/manage/Tiles/Description/Edit';
import EditTextTile from '@plone/volto/components/manage/Tiles/Text/Edit';
import EditImageTile from '@plone/volto/components/manage/Tiles/Image/Edit';
import EditVideoTile from '@plone/volto/components/manage/Tiles/Video/Edit';
import EditHeroImageLeftTile from '@plone/volto/components/manage/Tiles/HeroImageLeft/Edit';
import EditMapTile from '@plone/volto/components/manage/Tiles/Maps/Edit';
import EditHTMLTile from '@plone/volto/components/manage/Tiles/HTML/Edit';
import EditTableTile from '@plone/volto/components/manage/Tiles/Table/Edit';

import descriptionSVG from '@plone/volto/icons/description.svg';
import titleSVG from '@plone/volto/icons/text.svg';
import textSVG from '@plone/volto/icons/subtext.svg';
import cameraSVG from '@plone/volto/icons/camera.svg';
import videoSVG from '@plone/volto/icons/videocamera.svg';
import globeSVG from '@plone/volto/icons/globe.svg';
import codeSVG from '@plone/volto/icons/code.svg';
import heroSVG from '@plone/volto/icons/hero.svg';
import tableSVG from '@plone/volto/icons/table.svg';

defineMessages({
  title: {
    id: 'title',
    defaultMessage: 'Title',
  },
  description: {
    id: 'description',
    defaultMessage: 'Description',
  },
  text: {
    id: 'text',
    defaultMessage: 'Text',
  },
  image: {
    id: 'image',
    defaultMessage: 'Image',
  },
  video: {
    id: 'video',
    defaultMessage: 'Video',
  },
  hero: {
    id: 'hero',
    defaultMessage: 'Hero',
  },
  table: {
    id: 'table',
    defaultMessage: 'Table',
  },
  maps: {
    id: 'maps',
    defaultMessage: 'Maps',
  },
  html: {
    id: 'html',
    defaultMessage: 'HTML',
  },
  // Groups
  mostUsed: {
    id: 'mostUsed',
    defaultMessage: 'Most used',
  },
  media: {
    id: 'media',
    defaultMessage: 'Media',
  },
  common: {
    id: 'common',
    defaultMessage: 'Common',
  },
});

const groupTilesOrder = [
  { id: 'mostUsed', title: 'Most used' },
  { id: 'text', title: 'Text' },
  { id: 'media', title: 'Media' },
  { id: 'common', title: 'Common' },
];

const tilesConfig = {
  title: {
    id: 'title',
    title: 'Title',
    icon: titleSVG,
    group: 'text',
    view: ViewTitleTile,
    edit: EditTitleTile,
    restricted: true,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
  description: {
    id: 'description',
    title: 'Description',
    icon: descriptionSVG,
    group: 'text',
    view: ViewDescriptionTile,
    edit: EditDescriptionTile,
    restricted: true,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
  text: {
    id: 'text',
    title: 'Text',
    icon: textSVG,
    group: 'text',
    view: ViewTextTile,
    edit: EditTextTile,
    restricted: false,
    mostUsed: false,
    tileHasOwnFocusManagement: true,
    security: {
      addPermission: [],
      view: [],
    },
  },
  image: {
    id: 'image',
    title: 'Image',
    icon: cameraSVG,
    group: 'media',
    view: ViewImageTile,
    edit: EditImageTile,
    restricted: false,
    mostUsed: true,
    security: {
      addPermission: [],
      view: [],
    },
  },
  video: {
    id: 'video',
    title: 'Video',
    icon: videoSVG,
    group: 'media',
    view: ViewVideoTile,
    edit: EditVideoTile,
    restricted: false,
    mostUsed: true,
    security: {
      addPermission: [],
      view: [],
    },
  },
  hero: {
    id: 'hero',
    title: 'Hero',
    icon: heroSVG,
    group: 'common',
    view: ViewHeroImageLeftTile,
    edit: EditHeroImageLeftTile,
    restricted: false,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
  maps: {
    id: 'maps',
    title: 'Maps',
    icon: globeSVG,
    group: 'common',
    view: ViewMapTile,
    edit: EditMapTile,
    restricted: false,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
  html: {
    id: 'html',
    title: 'HTML',
    icon: codeSVG,
    group: 'common',
    view: ViewHTMLTile,
    edit: EditHTMLTile,
    restricted: false,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
  table: {
    id: 'table',
    title: 'Table',
    icon: tableSVG,
    group: 'common',
    view: ViewTableTile,
    edit: EditTableTile,
    restricted: false,
    mostUsed: false,
    security: {
      addPermission: [],
      view: [],
    },
  },
};

const requiredTiles = ['title'];

export { groupTilesOrder, requiredTiles, tilesConfig };
