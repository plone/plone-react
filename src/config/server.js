import imagesMiddleware from '@plone/volto/express-middleware/images';
import filesMiddleware from '@plone/volto/express-middleware/files';
import robotstxtMiddleware from '@plone/volto/express-middleware/robotstxt';
import sitemapMiddleware from '@plone/volto/express-middleware/sitemap';
import { readCriticalCss } from '@plone/volto/critical-css';

const settings = {
  expressMiddleware: [
    filesMiddleware(),
    imagesMiddleware(),
    robotstxtMiddleware(),
    sitemapMiddleware(),
  ],
  readCriticalCss,
};

export default settings;
