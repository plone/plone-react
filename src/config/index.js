/**
 * Config.
 * @module config
 */

import { defaultWidget, widgetMapping } from './Widgets';
import {
  layoutViews,
  contentTypesViews,
  defaultView,
  errorViews,
} from './Views';
import { nonContentRoutes } from './NonContentRoutes';
import ToHTMLRenderers, {
  options as ToHTMLOptions,
} from './RichTextEditor/ToHTML';
import {
  extendedBlockRenderMap,
  blockStyleFn,
  listBlockTypes,
} from './RichTextEditor/Blocks';
import plugins, { inlineToolbarButtons } from './RichTextEditor/Plugins';
import FromHTMLCustomBlockFn from './RichTextEditor/FromHTML';
import {
  groupBlocksOrder,
  requiredBlocks,
  blocksConfig,
  initialBlocks,
} from './Blocks';
import { loadables } from './Loadables';

import { sentryOptions } from './Sentry';
import { contentIcons } from './ContentIcons';

import applyAddonConfiguration from 'load-volto-addons';

const host = process.env.HOST || 'localhost';
const port = process.env.PORT || '3000';

const apiPath =
  process.env.RAZZLE_API_PATH ||
  (__DEVELOPMENT__
    ? `http://${host}:${port}/api`
    : 'http://localhost:8080/Plone');

let config = {
  settings: {
    host,
    port,
    // Internal proxy to bypass CORS *while developing*. Not intended for production use.
    // In production, the proxy is disabled, make sure you specify an apiPath that does
    // not require CORS to work.
    apiPath,
    devProxyToApiPath:
      process.env.RAZZLE_DEV_PROXY_API_PATH || 'http://localhost:8080/Plone', // Set it to '' for disabling the proxy
    // proxyRewriteTarget Set it for set a custom target for the proxy or overide the internal VHM rewrite
    // proxyRewriteTarget: '/VirtualHostBase/http/localhost:8080/Plone/VirtualHostRoot/_vh_api'
    // proxyRewriteTarget: 'https://myvoltositeinproduction.com'
    proxyRewriteTarget: process.env.RAZZLE_PROXY_REWRITE_TARGET || undefined,
    // apiPath: process.env.RAZZLE_API_PATH || 'http://localhost:8000', // for Volto reference
    // apiPath: process.env.RAZZLE_API_PATH || 'http://localhost:8081/db/web', // for guillotina
    actions_raising_api_errors: ['GET_CONTENT', 'UPDATE_CONTENT'],
    internalApiPath: process.env.RAZZLE_INTERNAL_API_PATH || undefined,
    websockets: process.env.RAZZLE_WEBSOCKETS || false,
    nonContentRoutes,
    extendedBlockRenderMap,
    blockStyleFn,
    listBlockTypes,
    FromHTMLCustomBlockFn,
    richTextEditorInlineToolbarButtons: inlineToolbarButtons,
    richTextEditorPlugins: plugins,
    ToHTMLRenderers,
    ToHTMLOptions,
    imageObjects: ['Image'],
    listingPreviewImageField: 'image',
    customStyleMap: null,
    notSupportedBrowsers: ['ie'],
    defaultPageSize: 25,
    isMultilingual: false,
    supportedLanguages: ['en'],
    defaultLanguage: 'en',
    navDepth: 1,
    expressMiddleware: [],
    defaultBlockType: 'text',
    verticalFormTabs: false,
    persistentReducers: ['blocksClipboard'],
    sentryOptions: {
      ...sentryOptions,
    },
    contentIcons: contentIcons,
    loadables,
    appExtras: [],
    maxResponseSize: 2000000000, // This is superagent default (200 mb)
  },
  widgets: {
    ...widgetMapping,
    default: defaultWidget,
  },
  views: {
    layoutViews,
    contentTypesViews,
    defaultView,
    errorViews,
  },
  blocks: {
    requiredBlocks,
    blocksConfig,
    groupBlocksOrder,
    initialBlocks,
  },

  addonRoutes: [],
  addonReducers: {},
};

config = applyAddonConfiguration(config);

export const settings = config.settings;
export const widgets = config.widgets;
export const views = config.views;
export const blocks = config.blocks;
export const addonRoutes = [...config.addonRoutes];
export const addonReducers = { ...config.addonReducers };
export const appExtras = config.appExtras;
