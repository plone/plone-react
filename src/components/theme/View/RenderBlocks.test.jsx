import React from 'react';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import RenderBlocks from './RenderBlocks';
import { render } from '@testing-library/react';

import config from '@plone/volto/registry';

const mockStore = configureStore();

test('Allows override of blocksConfig', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });

  const { container } = render(
    <Provider store={store}>
      <RenderBlocks
        blocksConfig={{
          ...config.blocks.blocksConfig,
          custom: {
            id: 'custom',
            view: ({ id, data }) => (
              <div>
                {id} - {data.text}
              </div>
            ),
          },
        }}
        content={{
          blocks_layout: {
            items: ['a', 'b'],
          },
          blocks: {
            a: {
              '@type': 'custom',
              text: 'a',
            },
            b: {
              '@type': 'custom',
              text: 'b',
            },
          },
        }}
      />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
