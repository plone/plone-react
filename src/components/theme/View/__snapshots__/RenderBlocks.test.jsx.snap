// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allows override of blocksConfig 1`] = `
<div>
  <div>
    <div>
      a
       - 
      a
    </div>
    <div>
      b
       - 
      b
    </div>
  </div>
</div>
`;
